subprojects/diagnostics/src/main/java/org/gradle/configuration/TaskDetailPrinter.java

```diff
@@ -41,48 +33,47 @@
 import java.util.Map;
 import java.util.Set;
 import java.util.TreeSet;
+import java.util.stream.Collectors;
 
 import static org.gradle.internal.logging.text.StyledTextOutput.Style.UserInput;
 import static org.gradle.util.internal.CollectionUtils.collect;
-import static org.gradle.util.internal.CollectionUtils.filter;
 import static org.gradle.util.internal.CollectionUtils.sort;
 
 @NonNullApi
 public class TaskDetailPrinter {
     private final String taskPath;
-    private final TaskSelection selection;
+    private final List<TaskDetails> tasks;
     private static final String INDENT = "     ";
-    private final OptionReader optionReader;
 
-    public TaskDetailPrinter(String taskPath, TaskSelection selection, OptionReader optionReader) {
+    public TaskDetailPrinter(String taskPath, Collection<TaskDetails> tasks) {
         this.taskPath = taskPath;
-        this.selection = selection;
-        this.optionReader = optionReader;
+        this.tasks = new ArrayList<>(tasks);
+        Collections.sort(this.tasks, new TaskDetails.Comparator());
```

Comment:
❓ Wouldn't it make sense to let the model store a sorted collection already?

Reply:
> ❓ Wouldn't it make sense to let the model store a sorted collection already?

I tend to think that is more of a presentation concern, which the printer is responsible for. Could be convinced otherwise.

Reply:
Again, motivation here was optimization